web/src/js/ducks/flows/index.ts

```diff
@@ -0,0 +1,366 @@
+        list[listPos] = flow;
+        const byId = new Map(state.byId);
+        byId.set(flow.id, flow);
+        // Update view
+        const prevViewPos: number | undefined = _viewIndex.get(flow.id);
+        const hasOldFlow = prevViewPos !== undefined;
+        const hasNewFlow = !(matching_filters[FilterName.Search] === false);
+        if (hasNewFlow && !hasOldFlow) {
+            ({ view, _viewIndex } = insertViewItem(
+                view,
+                _viewIndex,
+                flow,
+                makeSort(sort),
+            ));
+        } else if (!hasNewFlow && hasOldFlow) {
+            ({ view, _viewIndex } = removeViewItemAt(
+                view,
+                _viewIndex,
+                prevViewPos,
+            ));
+        } else if (hasNewFlow && hasOldFlow) {
+            ({ view, _viewIndex } = updateViewItem(
+                view,
+                _viewIndex,
+                flow,
+                makeSort(sort),
+            ));
+        }
+        // Update selection
+        if (selectedIds.has(flow.id)) {
+            selected = selected.map((existing) =>
+                existing.id === flow.id ? flow : existing,
+            );
+        }
+        // Update highlight
+        if (matching_filters[FilterName.Highlight]) {
+            if (!highlighted.has(flow.id)) {
+                highlighted = new Set(highlighted);
+                highlighted.add(flow.id);
+            }
+        } else {
+            highlighted = withElemRemoved(highlighted, flow.id);
+        }
+
+        return {
+            list,
+            _listIndex,
+            byId,
+            view,
+            _viewIndex,
+            sort,
+            selected,
+            selectedIds,
+            highlighted,
+        };
+    } else if (FLOWS_REMOVE.match(action)) {
+        const flow_id = action.payload;
+        const { sort } = state;
+        let { view, _viewIndex, selected, selectedIds } = state;
+        const listPos = state._listIndex.get(flow_id)!;
+        const list = state.list.toSpliced(listPos, 1);
+        const _listIndex = withKeyRemoved(state._listIndex, flow_id);
+        const byId = withKeyRemoved(state.byId, flow_id);
+        // Update view
+        const viewPos = _viewIndex.get(flow_id);
+        if (viewPos !== undefined) {
+            ({ view, _viewIndex } = removeViewItemAt(
+                view,
+                _viewIndex,
+                viewPos,
+            ));
+        }
+        // Update selection
+        if (selectedIds.has(flow_id)) {
+            if (selected.length === 1 && viewPos !== undefined) {
+                const fallback =
+                    view[viewPos /* no +1, already removed */] ??
+                    view[viewPos - 1];
+                selected = [fallback];
+            } else {
+                selected = selected.filter((f) => f.id !== flow_id);
+            }
+            selectedIds = buildLookup(selected);
+        }
```

Comment:
There is a problem here that arises when removing the last flow. Basically selected is equal to `[undefined]` and `buildLookup(selected)` crashes.
I think this check might help: `selected = fallback ? [fallback] : []`, but it doesn't fully solve the problem. If you try to clear all the flows, we no longer see the `mitmproxy is running...` message, which suggests that the `list` of flows is not actually empty and still contains elements. I don't have time to investigate further right now, but I can take a closer look later today or tomorrow morning.

Reply:
Fixed somewhere in https://github.com/mitmproxy/mitmproxy/pull/7704/commits/bd7b35c7885bc1f25ba307af884205b59e0bcfc1! :)